### PR TITLE
Add U2F Support to Sudo

### PR DESCRIPTION
**Pull request by mitsuhiko**
This pull request improves the u2f internals and adds U2F support as an
alternative to the sudo screen.

This requires changes in the sudo module: https://github.com/mattrobenolt/django-sudo/pull/12

### CODE REVIEW

**Comment by mitsuhiko** [comment]
@getsentry/infrastructure requires review.

**Comment by mitsuhiko** [comment]
(The tests fail because of the sudo dependency)

File: src/sentry/models/authenticator.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def auto_add_recovery_codes(self, user, force=False):
             ):
                 iface = authenticator.interface
                 if iface.is_backup_interface:
-                    return iface
+                    return
```
Comment (mattrobenolt) [inline src/sentry/models/authenticator.py]: Was this the bug you poked me about?

**Comment by mattrobenolt** [comment]
Python seems chill, but would like to get some @benvinegar eyes on it since it's mostly js changes.

**Comment by mitsuhiko** [comment]
Tests fixed now.

File: src/sentry/static/sentry/app/components/u2finterface.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      return null;
+    }
+    return (
+      <div className="u2f-box">
+        <p className="error">
+          {t(`
+           Unfortunately your browser does not support U2F. You need to use
+           a different two-factor method or switch to a browser that supports
+           it (Google Chrome or Microsoft Edge).
+          `)}
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/u2finterface.jsx]: Does leading/trailing whitespace get trimmed when looking up the translation string?

File: src/sentry/static/sentry/app/components/u2finterface.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        <input type="hidden" name="challenge" ref={this.bindChallengeElement}/>
+        <input type="hidden" name="response" ref={this.bindResponseElement}/>
+        {this.props.children}
+      </div>
+    );
+  },
+
+  render() {
+    let {isSupported} = this.state;
+    if (isSupported === null) {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/u2finterface.jsx]: When can `isSupported` be null (vs falsy)? Worth commenting?

File: src/sentry/static/sentry/app/components/u2finterface.jsx (hunk 2 above)
Comment (mitsuhiko) [inline src/sentry/static/sentry/app/components/u2finterface.jsx]: Nope. But we have fuzzy matching on updating translations.

**Pull request merged**